File: stuff_data/src/main/java/android/montreal/devfest/data/SharedPrefsKeyValueStore.java
package android.montreal.devfest.data;

import android.content.Context;
import android.content.SharedPreferences;
import android.montreal.devfest.stuff.data.gateway.StuffStorage;
import android.preference.PreferenceManager;

class SharedPrefsKeyValueStore implements StuffStorage {
    private final SharedPreferences sharedPreferences;

    public SharedPrefsKeyValueStore(Context context) {
        this.sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
    }

    @Override
    public void setString(String key, String value) {
        this.sharedPreferences.edit().putString(key, value).apply();
    }

    @Override
    public String getString(String key) {
        return this.sharedPreferences.getString(key, null);
    }

    @Override
    public void setInt(String key, int value) {
        this.sharedPreferences.edit().putInt(key, value).apply();
    }

    @Override
    public int getInt(String key) {
        return this.sharedPreferences.getInt(key, 0);
    }
}

File: stuff_data/src/main/java/android/montreal/devfest/stuff/data/SharedPrefsKeyValueStore.kt
package android.montreal.devfest.stuff.data

import android.content.Context
import android.content.SharedPreferences
import android.montreal.devfest.stuff.data.gateway.StuffStorage
import android.preference.PreferenceManager

internal class SharedPrefsKeyValueStore(context: Context) : StuffStorage {
    private val sharedPreferences: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)

    override fun setString(key: String, value: String?) {
        this.sharedPreferences.edit().putString(key, value).apply()
    }

    override fun getString(key: String): String? {
        return this.sharedPreferences.getString(key, null)
    }

    override fun setInt(key: String, value: Int) {
        this.sharedPreferences.edit().putInt(key, value).apply()
    }

    override fun getInt(key: String): Int {
        return this.sharedPreferences.getInt(key, 0)
    }
}
